command/agent/command.go

```diff
@@ -135,6 +136,12 @@ func (c *Command) readConfig() *Config {
 			return nil
 		}
 
+		// The user asked us to load some config here but we didn't find any,
+		// so we'll complain but continue.
+		if current == nil || reflect.DeepEqual(current, &Config{}) {
```

Comment:
That case will come back as an error. The specific case this handles is when the user specifies a directory and that directory does not contain any `.hcl` or `.json` files. In that case the LoadConfigDir call returns `&Config{}` and the command fails silently.

If we prefer we can change that case to an error.